### DISCUSSION TITLE
Bad network performance with multiple robots in ROS2

### DISCUSSION THREAD

**Discussion by Krzo99** [eProsima/Fast-DDS#4948] [Q&A] [answered]
Hello,

I'm using ROS2 with FastDDS. With one robot (and the PC), the network works well, without any problems.
However, as soon as I add another robot, running the same software as the first, the network grinds to a halt, 90% of the time.
This means, msgs not being received on the PC, or being 10s-15s late. There isn't any topic name collisions, as the same setup works the other 10% of the time. 

Each robot has hundreds of topics, publishers & subscribers on ~10 nodes. My theory is, that the network gets overwhelmed with all the multicasting, and grinds the network to a halt, creating a multicast storm. Sometimes the router even crashes/reboots.

Would it be possible to configure fastdds_config.xml file, to disable multicasting, if I know all the IP addresses of the robots & PC.
Could you provide the example, I'm a bit lost,
PCs IP is 192.168.1.123, robot's IP is: 192.168.1.111

Thank you for all the help!

**Comment by jonatansw** [comment]
Hi Krzo99,

Unfortunatelly not able to help, but we are experiencing the very same thing, especially with images hardly working at all.
Did you make any progress on this?

Thanks

**Reply by Krzo99** [reply]
We used seperate DOMAIN_ID's for each robot, and connected only important topics with the domain_bridge package. Maybe JesusPoderoso's answer might help you.

**Comment by JesusPoderoso** [comment]
Hi @Krzo99, @jonatansw, thanks for using Fast DDS.
In deployments where the amount of nodes is really small, the [initial peers](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/simple.html#initial-peers) configuration is recommended.
It would imply adding the IP address of the participants to which the communication wants to be established, per participant. 
The [Discovery Server](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery_server.html) mechanism would also reduce significantly the network traffic caused by multicast, and it is more scalable than initial peers. In this case, one participant would act as a server (PC) and the other participant(s) (robots) would connect to the server that acts as a hub for meta traffic discovery.
The following link refers to a [Discovery Server Vulcanexus tutorial](https://docs.vulcanexus.org/en/latest/rst/tutorials/core/discoveryserver/discoveryserverminimalexample/discovery_server_minimal_example.html). The [Use ROS 2 with Fast-DDS Discovery Server](https://fast-dds.docs.eprosima.com/en/latest/fastdds/ros2/discovery_server/ros2_discovery_server.html) documentation section may be of your interest too, and the C++ and XML configuration is also explained in the [Discovery Server section](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery_server.html) of the Fast DDS documentation.
Hope it helps!

**Comment by Krzo99** [comment, marked as answer]
In the end, I solved it by using seperate DOMAIN_ID's for every robot, and connecting them using domain_bridge package. It works well, but feels like a hack.
